src/Microsoft.AspNetCore.ResponseCompression/ResponseCompressionOptions.cs

```diff
@@ -15,9 +15,14 @@ public class ResponseCompressionOptions
         /// </summary>
         public IEnumerable<string> MimeTypes { get; set; }
 
+        /// <summary>
+        /// Configures which MIME types will be compressed.
+        /// </summary>
+        public IMimeTypeFilter MimeTypeFilter { get; set; }
```

Comment:
Putting an interface onto Options is going to be an maintenance barrier. We won't be able to add to or modify the interface without breaking implementers. One way of dealing with this is to remove the interface and use a concrete type, making sure everything is virtual so it can be overridden.